x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/aggs/changepoint/ChangePointAggregator.java

```diff
@@ -40,53 +29,14 @@ public class ChangePointAggregator extends SiblingPipelineAggregator {
     private static final Logger logger = LogManager.getLogger(ChangePointAggregator.class);
 
     static final double P_VALUE_THRESHOLD = 0.01;
-    private static final int MINIMUM_BUCKETS = 10;
-    private static final int MAXIMUM_CANDIDATE_CHANGE_POINTS = 1000;
-    private static final int MAXIMUM_SAMPLE_SIZE_FOR_KS_TEST = 500;
-    private static final KolmogorovSmirnovTest KOLMOGOROV_SMIRNOV_TEST = new KolmogorovSmirnovTest();
+    static final int MINIMUM_BUCKETS = 10;
 
     private static double changePValueThreshold(int nValues) {
```

Comment:
Maybe this should be on `ChangeDetector` too, but with `P_VALUE_THRESHOLD` supplied as `minValueCountPValue`.

Reply:
I agree. At first I didn't realize it's only applicable for change detection. Moved it now